src/x86_64/Ginit.c

```diff
@@ -142,18 +142,18 @@ write_validate (void *addr)
        /* use syscall insteadof write() so that ASAN does not complain */
        ret = syscall (SYS_write, mem_validate_pipe[1], addr, 1);
 #else
-	  ret = write (mem_validate_pipe[1], addr, 1);
+	  ret = write (mem_validate_pipe[1], (void *)addr, 1);
 #endif
     }
   while ( errno == EINTR );
 
   return ret;
 }
 
-static int (*mem_validate_func) (void *addr, size_t len);
-static int msync_validate (void *addr, size_t len)
+static int (*mem_validate_func) (unw_word_t addr, size_t len);
+static int msync_validate (unw_word_t addr, size_t len)
 {
-  if (msync (addr, len, MS_ASYNC) != 0)
+  if (msync ( (void *)addr, len, MS_ASYNC) != 0)
```

Comment:
It looks like msync needs to be page aligned also?

man msync "EINVAL addr is not a multiple of PAGESIZE"

Reply:
Ah, I'd even checked the manpage, but missed the reference in the ERRORS section

I have fixed the problem, and added an env var to force the validation function. This allowed me to test it manually at least. If you'd rather I removed env var toggle from the production code, let me know.